### DISCUSSION TITLE
Q: is it possible to plot a msh in a juypyter notebook?

### DISCUSSION THREAD

**Discussion by ggrrll** [simnibs/simnibs#137] [9. Results and analysis] [answered]
I have tried with backend like `ipympl` but did not work  (no plot appears)

(in that way , it would be  faster to write reports, instead of saving and importing a screenshot every time )

thanks

**Comment by jduemose** [comment, marked as answer]
Hi

One possibility is to use `pyista` which also supports interactive backends for notebooks, e.g., https://docs.pyvista.org/version/stable/user-guide/jupyter/trame.html#trame-jupyter.

Below plots the gray matter surface. This uses the convenience function `make_tri_mesh` but you can easily construct tetrahedral meshes as well (and slice them).
```
import pyvista as pv
import simnibs

GM_SURFACE = 1002
m = simnibs.read_msh("/path/to/mesh.msh")
c = m.crop_mesh(GM_SURFACE)

mesh = pv.make_tri_mesh(
    c.nodes.node_coord,
    c.elm.node_number_list[:,:3]-1,
)
# If you have elm/node data that you want to plot on the surface, then something like this
# mesh["magnE"] = c.elmdata[1].value

# and plot
mesh.plot()
# or for slightly more control
# p = pv.Plotter()
# p.add_mesh(mesh)
# p.show()
```

**Comment by ggrrll** [comment]
thanks 

I have been trying but the kernel keeps dying  :/ -- still haven't figure out why

**Reply by ggrrll** [reply]
following https://github.com/pyvista/pyvista/discussions/1847

**Reply by ggrrll** [reply]
ok, fixed the kernel issue by installing `libgl1-mesa-glx`
